libbeat/autodiscover/template/config.go

```diff
@@ -45,14 +45,19 @@ type MapperSettings []*struct {
 func NewConfigMapper(configs MapperSettings) (*Mapper, error) {
 	var mapper Mapper
 	for _, c := range configs {
-		condition, err := conditions.NewCondition(c.ConditionConfig)
-		if err != nil {
-			return nil, err
+		condMap := &ConditionMap{Configs: c.Configs}
+
+		if c.ConditionConfig == nil {
```

Comment:
You could be doing this `nil` check here: https://github.com/elastic/beats/blob/6d0704f32a14b298b1dc2a52f51bb00d2841739f/libbeat/autodiscover/template/config.go#L83, then you don't need the overhead of adding a new bool condition to the `conditions` package